storage/replica.go

```diff
@@ -1111,12 +1112,27 @@ func (r *Replica) handleRaftReady() error {
 	// but it's hard to distinguish so we resubmit everything
 	// anyway). We delay resubmission until after we have processed
 	// the entire batch of entries.
-	hasEmptyEntry := false
+	reproposeCmds := func() error { return nil }
 	for _, e := range rd.CommittedEntries {
 		switch e.Type {
 		case raftpb.EntryNormal:
 			if len(e.Data) == 0 {
-				hasEmptyEntry = true
+				reproposeCmds = func() error {
```

Comment:
A non-constant function pointer is not a common pattern, and I like the old style with `hasEmptyEntry` better (it may also be faster - does this allocate a new closure object each time the assignment occurs?). Use an inline function in the `if hasEmptyEntry` block but don't move the code up here.